odoo/orm/domains.py

```diff
@@ -1088,6 +1088,22 @@ def _optimize_in_list(condition, _):
     return DomainCondition(condition.field_expr, condition.operator, OrderedSet(value))
 
 
+@operator_optimization(['in', 'not in'])
+def _optimize_in_required(condition, model):
```

Comment:
Operator optimizations are part of the `BASE` optimization level.  But this one depends on the database's state, i.e., it is transaction-dependent.  IMHO this optimization should be part of the `FOR_SQL` optimization level.

Reply:
Strictly speaking yes. However, this does not change within the lifetime of the Registry (it depends on the schema, but does not depend on the data/state) and if it is saved in some cache, it will be purged after the registry reload.

Should we keep it as-is or can  you confirm to move this in the FOR_SQL optimizations?